providers/defltprov.c

```diff
@@ -366,6 +366,7 @@ static const OSSL_ALGORITHM deflt_signature[] = {
 #ifndef OPENSSL_NO_DSA
     { "DSA:dsaEncryption", "default=yes", dsa_signature_functions },
 #endif
+    { "RSA:rsaEncryption", "default=yes", rsa_signature_functions },
```

Comment:
I'm confused, I thought that was to be treated as a separate algorithm...  (that's what the EVP_PKEY_METHOD does, even though it ends up calling very much the same functions otherwise).

Reply:
Hmmm. Well yes. And no. As you discovered you can do pss signatures (or just pkcs1 signatures) with an an RSA key with the "rsaEncryption" OID. But you can't do asym encryption with a key with the pss oid. But as far as this code is concerned you can sign with either. With the asym cipher code if you attempt to do an encryption operation with a PSS pad mode it will fail.

Reply:
The thing is that we *must* keep them separate, or there will be hell to pay later, when we attempt to wrap RSA-PSS keys in PKCS#8.  Making that an alias for RSA will lead to all kinds of strange results.

Reply:
Wouldn't aliasing them also mean that the ASYM_CIPHER RSA implementation suddenly be called for RSA-PSS keys, when it really shouldn't?

Reply:
I have in my mind that I would do a sweep over all our provided RSA implementations and update for PSS where needed.  Do note that the KEYMGMT implementation has poor PSS support for the moment.

Reply:
> The thing is that we must keep them separate, or there will be hell to pay later, when we attempt to wrap RSA-PSS keys in PKCS#8. Making that an alias for RSA will lead to all kinds of strange results.

So do we just need a duplicate line here with the PSS names, but still pointing at the same signature functions (and similarly for the KEYMGMT line)?

Reply:
> I have in my mind that I would do a sweep over all our provided RSA implementations and update for PSS where needed. Do note that the KEYMGMT implementation has poor PSS support for the moment.

Do you see that as a separate PR? I could accept that as an answer.

Reply:
I see that as a separate PR, yes.

> So do we just need a duplicate line here with the PSS names, but still pointing at the same signature functions (and similarly for the KEYMGMT line)?

Not quite.  The newctx or init functions needs to be different.  And that's my main reason to want to make that a separate effort.